assets/src/edit-story/components/library/panes/pageLayouts/pageLayouts.js

```diff
@@ -101,33 +225,49 @@ function PageLayouts(props) {
       }}
     >
       <PageLayoutsContainer height={rowVirtualizer.totalSize}>
-        {rowVirtualizer.virtualItems.map((virtualRow) => {
-          const firstColumnIndex = virtualRow.index * 2;
-          const indexes = [firstColumnIndex, firstColumnIndex + 1];
-          return (
-            <PageLayoutsRow
-              key={virtualRow.index}
-              height={virtualRow.size}
-              translateY={virtualRow.start}
-            >
-              {indexes.map((index) => {
-                const page = pages[index];
-                if (!page) {
-                  return null;
-                }
-                return (
-                  <PageLayout
-                    key={page.id}
-                    page={page}
-                    pageSize={pageSize}
-                    onConfirm={() => handleApplyPageLayout(page)}
-                    requiresConfirmation={requiresConfirmation}
-                  />
-                );
-              })}
-            </PageLayoutsRow>
-          );
-        })}
+        <PageLayoutsVirtualizedContainer
+          height={rowVirtualizer.totalSize}
+          ref={containerRef}
+          pageSize={pageSize}
+          tab={0}
+          onFocus={handlePageLayoutFocus}
+        >
+          {rowVirtualizer.virtualItems.map((virtualRow) =>
+            columnVirtualizer.virtualItems.map((virtualColumn) => {
+              const pageIndex =
+                virtualColumn.index === 0
+                  ? virtualRow.index * 2
+                  : virtualRow.index * 2 + 1;
+              const page = pages[pageIndex];
+
+              if (!page) {
+                return null;
+              }
+              const isActive = activePageId === page.id && isPageLayoutsFocused;
+
+              return (
+                <PageLayout
+                  key={pageIndex}
+                  ref={(el) => (pageRefs.current[page.id] = el)}
+                  translateY={virtualRow.start}
+                  translateX={virtualColumn.start}
+                  page={page}
+                  pageSize={pageSize}
+                  isActive={isActive}
+                  onFocus={() => handlePageFocus(page.id)}
+                  onClick={() => handlePageClick(page)}
+                  onKeyUp={(event) => handleKeyboardPageClick(event, page)}
+                />
+              );
+            })
+          )}
+        </PageLayoutsVirtualizedContainer>
+        {isConfirming && (
```

Comment:
This is better here. I didn't do this because I didn't have any reason to track active page before, but now that we need it 👏 .